pkg/apis/core/validation/validation_patch.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	apimachinerytypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+var (
+	// we make an exception for the following secret objects, during update
```

Comment:
Maybe it's worth explaining why we are allowing for this exception. We have multiple controllers reconciling the same object, resulting in unexpected outcomes such as the generation of new key pairs. Our goal is to prevent the generation of new key pairs by disallowing deletions and permitting only updates, which appear to be 'safe'.